rllib/core/optim/tests/test_rl_optimizer.py

```diff
@@ -0,0 +1,378 @@
+            next_obs, reward, done, _ = env.step(action)
+            _next_obs.append(next_obs)
+            _actions.append([action])
+            _rewards.append([reward])
+            _dones.append([done])
+            _return += reward
+            if done:
+                break
+            obs = next_obs
+        batch = SampleBatch(
+            {
+                "obs": _obs,
+                "next_obs": _next_obs,
+                "actions": _actions,
+                "rewards": _rewards,
+                "dones": _dones,
+            }
+        )
+        _returns.append(_return)
+        _rollouts.append(batch)
+    return np.mean(_returns), _returns, _rollouts
+
+
+class BCTorchModule(TorchRLModule):
+    def __init__(self, config: FCConfig) -> None:
+        super().__init__(config)
+        self.policy = FCNet(config)
+
+    def input_specs(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+                "actions": TorchTensorSpec("b"),
+            }
+        )
+
+    def output_specs(self) -> ModelSpec:
+        return ModelSpec({"action_dist": Distribution})
+
+    def input_specs_exploration(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_inference(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_train(self) -> ModelSpec:
+        return self.input_specs()
+
+    def output_specs_exploration(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_inference(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_train(self) -> ModelSpec:
+        return self.output_specs()
+
+    def _forward_inference(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        with torch.no_grad():
+            action_logits = self.policy(obs)
+        action_logits_inference = torch.argmax(action_logits, dim=-1)
+        action_dist = TorchDeterministic(action_logits_inference)
+        return {"action_dist": action_dist}
+
+    def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
+        with torch.no_grad():
+            return self._forward_train(batch)
+
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+        action_dist = TorchCategorical(logits=action_logits)
+        return {"action_dist": action_dist}
+
+    def is_distributed(self) -> bool:
+        return False
+
+    def make_distributed(self, dist_config: Mapping[str, Any] = None) -> None:
+        pass
+
```

Comment:
Let's leave these api's implementations to the parent class to save us from trouble later. Those are yet to be determined and you are not adding any new behavior to them here anyway. So might be better to not do anything about them in this unittest.